packages/backend-core/src/db/index.js

```diff
@@ -1,13 +1,64 @@
-let Pouch
+const pouch = require("./pouch")
 
-module.exports.setDB = pouch => {
-  Pouch = pouch
+let PouchDB
+let initialised = false
+
+const put =
+  dbPut =>
+  async (doc, options = {}) => {
+    const response = await dbPut(doc, options)
+    // TODO: add created / updated
+    return response
+  }
+
+const checkInitialised = () => {
+  if (!initialised) {
+    throw new Error("init has not been called")
+  }
+}
+
+exports.init = opts => {
+  PouchDB = pouch.getPouch(opts)
+  initialised = true
+}
+
+// NOTE: THIS IS A DANGEROUS FUNCTION - USE WITH CAUTION
+// this function is prone to leaks, should only be used
+// in situations that using the function doWithDB does not work
+exports.dangerousGetDB = (dbName, opts) => {
+  checkInitialised()
+  const db = new PouchDB(dbName, opts)
+  const dbPut = db.put
+  db.put = put(dbPut)
+  return db
+}
+
+// use this function if you have called dangerousGetDB - close
+// the databases you've opened once finished
+exports.closeDB = async db => {
+  if (!db) {
+    return
+  }
+  try {
+    return db.close()
+  } catch (err) {
+    // ignore error, already closed
+  }
 }
 
-module.exports.getDB = dbName => {
-  return new Pouch(dbName)
+// we have to use a callback for this so that we can close
+// the DB when we're done, without this manual requests would
+// need to close the database when done with it to avoid memory leaks
+exports.doWithDB = async (dbName, cb, opts) => {
+  const db = exports.dangerousGetDB(dbName, opts)
+  // need this to be async so that we can correctly close DB after all
+  // async operations have been completed
+  const resp = await cb(db)
+  await exports.closeDB(db)
```

Comment:
Should this do something like:
```js
try {
   const resp = await cb(db)
   return resp
} finally {
   await exports.closeDB(db)
}
```

Similar to how the closing is done in the `doInAppContext` and `doInTenant`,  we'd get all the manual usages covered then as well.